core/01_core.js

```diff
@@ -238,40 +239,44 @@
     opCallTraces,
     refOp,
     unrefOp,
-    close: opSync.bind(null, "op_close"),
-    tryClose: opSync.bind(null, "op_try_close"),
+    close: (rid) => unwrapOpResult(ops.op_close(rid)),
+    tryClose: (rid) => unwrapOpResult(ops.op_try_close(rid)),
     read: opAsync.bind(null, "op_read"),
     write: opAsync.bind(null, "op_write"),
     shutdown: opAsync.bind(null, "op_shutdown"),
-    print: opSync.bind(null, "op_print"),
-    setMacrotaskCallback: opSync.bind(null, "op_set_macrotask_callback"),
-    setNextTickCallback: opSync.bind(null, "op_set_next_tick_callback"),
-    runMicrotasks: opSync.bind(null, "op_run_microtasks"),
-    hasTickScheduled: opSync.bind(null, "op_has_tick_scheduled"),
-    setHasTickScheduled: opSync.bind(null, "op_set_has_tick_scheduled"),
-    evalContext: opSync.bind(null, "op_eval_context"),
-    createHostObject: opSync.bind(null, "op_create_host_object"),
-    encode: opSync.bind(null, "op_encode"),
-    decode: opSync.bind(null, "op_decode"),
-    serialize: opSync.bind(null, "op_serialize"),
-    deserialize: opSync.bind(null, "op_deserialize"),
-    getPromiseDetails: opSync.bind(null, "op_get_promise_details"),
-    getProxyDetails: opSync.bind(null, "op_get_proxy_details"),
-    isProxy: opSync.bind(null, "op_is_proxy"),
-    memoryUsage: opSync.bind(null, "op_memory_usage"),
-    setWasmStreamingCallback: opSync.bind(
-      null,
-      "op_set_wasm_streaming_callback",
-    ),
-    abortWasmStreaming: opSync.bind(null, "op_abort_wasm_streaming"),
-    destructureError: opSync.bind(null, "op_destructure_error"),
-    terminate: opSync.bind(null, "op_terminate"),
-    opNames: opSync.bind(null, "op_op_names"),
-    eventLoopHasMoreWork: opSync.bind(null, "op_event_loop_has_more_work"),
-    setPromiseRejectCallback: opSync.bind(
-      null,
-      "op_set_promise_reject_callback",
-    ),
+    print: (msg, isErr) => unwrapOpResult(ops.op_print(msg, isErr)),
+    setMacrotaskCallback: (fn) =>
+      unwrapOpResult(ops.op_set_macrotask_callback(fn)),
+    setNextTickCallback: (fn) =>
+      unwrapOpResult(ops.op_set_next_tick_callback(fn)),
+    runMicrotasks: () => ops.op_run_microtasks(),
+    hasTickScheduled: () => ops.op_has_tick_scheduled(),
```

Comment:
```suggestion
    runMicrotasks: ops.op_run_microtasks,
    hasTickScheduled: ops.op_has_tick_scheduled,
```
?
(and for others?)

Reply:
`ops` cannot actually be directly referenced. It seems that this JS run even when Deno isn't really being set up, eg. in tests or such I think.

In (some?) of those cases `ops` will be undefined and these sorts of lines will throw an error.